File: 1822: signOfProduct.cpp
// problem : https://leetcode.com/problems/sign-of-the-product-of-an-array/description/

class Solution {
public:
    int arraySign(vector<int>& nums) {
        
        bool neg = false;
        for(int i  = 0; i < nums.size(); i++){

            if(nums[i] == 0){

                return 0;
            }

            if(nums[i] < 0){

                flip(neg);
            }
        }

        if(neg){

            return -1;
        }

        return 1;


    }

    void flip(bool &x){

        if(x){

            x = false;
        }
        else{

            x = true;
        }
        
    }


};

File: 1822: signOfProduct.py
class Solution:
    def arraySign(self, nums: List[int]) -> int:
        m = {}
        m[1] = 0
        m[-1] = 0
        for x in nums:
            if x == 0:
                return 0
            elif x > 0:
                m[1] = m[1]+1
            else:
                m[-1] = m[-1]+1
        if (m[-1]%2) == 1:
            return -1
        return 1
        